### ISSUE TITLE
Lines wrap incorrectly when followed by a non-wrappable line

### ISSUE THREAD

**Issue by edwardloveall** [kivikakk/comrak#227]
Hello! I believe I've found a bug. I have a markdown file `untitled.md`:

```md
item that doesn't need wrapping

a-long-line-that-won't-be-wrapped-because-there-is-no-character-it-can-break-on
```

When running this through the cli with the `--to commonmark` and `--width 72` options:

```sh
comrak --to commonmark --width 72 untitled.md
```

It outputs the following:

```md
item that doesn't need
wrapping

a-long-line-that-won't-be-wrapped-because-there-is-no-character-it-can-break-on
```

I think the list item is not supposed to wrap there. If the last line is short enough to not need wrapping, this doesn't occur. If the last line is wrappable (e.g. remove all the `-` characters), this also doesn't occur.

**Comment by kivikakk** [comment]
Thanks for this report — verified. Will look into a fix in a little bit.

Also, if you felt like giving it a go yourself and had the time, please feel free! Should be a nice easy one to just compare the wrapping behaviour of comrak's CommonMark formatter and cmark-gfm's.

**Comment by edwardloveall** [comment]
Cool. I made an attempt at a fix in #228
